pkg/cli/debug/debug.go

```diff
@@ -785,6 +785,7 @@ func (o *DebugOptions) transformPodForDebug(annotations map[string]string) (*cor
 		zero := int64(0)
 		container.SecurityContext.RunAsUser = &zero
 		container.SecurityContext.RunAsNonRoot = nil
+		pod.Spec.SecurityContext.RunAsNonRoot = nil
```

Comment:
According to docs either this or the above should be set as such. Can you verify if both are present which takes precedence, it's possible that we'll need to apply this rule not just here but in the accompanying places accordingly.

Reply:
When updating please add a comment explaining why we're doing so.

Reply:
According to doc, if set in both Container SecurityContext and Pod SecurityContext, the value specified in Container SecurityContext takes precedence.
So we need to take care of cases where Pod SecurityContext has some value and `oc debug` set Container SecurityContext to nil.